### ISSUE TITLE
CA1416 ("Validate platform compatibility") ignores `[DoesNotReturn]`

### ISSUE THREAD

**Issue by LukasGelke** [dotnet/roslyn-analyzers#7345]
### Analyzer

**Diagnostic ID**: [CA1416 ](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1416): `Validate platform compatibility`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: `8.0.0` and `9.0.0-preview.24225.1`

### Describe the bug

While making our internal Projects Linux-Compatible (or at last starting), we are compiling one single "general" net6 Assembly for both "windows" and "linux" Platforms.
While writing tests (MsTest v3) `Assert.Inconclusive()` is marked with `[DoesNotReturn]`. However, it does not get treated properly.

### Steps To Reproduce

```cs
internal class MyClass
{
  public void Do()
  {
    if (!OperatingSystem.IsWindows())
    {
      ThrowEx();
      // return;
      // throw new InvalidOperationException();
    }
    WindowsMethod(); // CA1416 here
  }

  [SupportedOSPlatform("windows")]
  private static void WindowsMethod() { }

  [DoesNotReturn]
  private static void ThrowEx() => throw new InvalidOperationException();
}
```

### Expected behavior
Methods with `[DoesNotReturn]` gets treated like an "inline exception"

### Actual behavior
Analyzer complains at `WindowsMethod()`

**Comment by buyaa-n** [comment]
Well `[DoesNotReturn]` was not really expected to be treated like an "inline exception", so it is not a bug.

Plis I think the methods with `[DoesNotReturn]` attribute could have a platform dependent logic before throwing
